shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobPersistService.java

```diff
@@ -0,0 +1,120 @@
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+
+/**
+ * Rule altered job persist service.
+ */
+
+@Slf4j
+public final class RuleAlteredJobPersistService {
+    
+    private static final Map<String, Map<Integer, JobPersistIntervalParameter>> JOB_PERSIST_MAP = new ConcurrentHashMap<>();
+    
+    private static final GovernanceRepositoryAPI REPOSITORY_API = PipelineAPIFactory.getGovernanceRepositoryAPI();
+    
+    private static final ScheduledExecutorService JOB_PERSIST_EXECUTOR = Executors.newSingleThreadScheduledExecutor(ExecutorThreadFactoryBuilder.build("scaling-job-schedule-%d"));
+    
+    static {
+        JOB_PERSIST_EXECUTOR.scheduleWithFixedDelay(new PersistJobContextRunnable(), 5, 1, TimeUnit.SECONDS);
+    }
+    
+    /**
+     * Remove job schedule map by job id.
+     *
+     * @param jobId job id
+     */
+    public static void removeJobPersistParameter(final String jobId) {
+        log.info("Remove job persist, job id: {}", jobId);
+        JOB_PERSIST_MAP.remove(jobId);
+    }
+    
+    /**
+     * Add job schedule.
+     *
+     * @param jobId job id
+     * @param shardingItem sharding item
+     */
+    public static void addJobPersistParameter(final String jobId, final int shardingItem) {
+        log.info("Add job schedule, jobId={}, shardingItem={}", jobId, shardingItem);
+        JOB_PERSIST_MAP.computeIfAbsent(jobId, key -> new ConcurrentHashMap<>()).put(shardingItem, new JobPersistIntervalParameter(jobId, shardingItem));
+    }
+    
+    /**
+     * Persist job process, may not be implemented immediately, depending on persist interval.
+     *
+     * @param jobId job id
+     * @param shardingItem sharding item
+     */
+    public static void triggerPersist(final String jobId, final int shardingItem) {
+        Map<Integer, JobPersistIntervalParameter> intervalParamMap = JOB_PERSIST_MAP.getOrDefault(jobId, Collections.emptyMap());
+        JobPersistIntervalParameter parameter = intervalParamMap.get(shardingItem);
+        if (null == parameter) {
+            log.debug("Persist interval parameter is null, jobId={}, shardingItem={}", jobId, shardingItem);
+            return;
+        }
+        parameter.getAlreadyPersisted().compareAndSet(true, false);
+    }
+    
+    private static void persist(final String jobId, final int shardingItem, final long persistTimeMillis, final JobPersistIntervalParameter param) {
+        Map<Integer, RuleAlteredJobScheduler> schedulerMap = RuleAlteredJobSchedulerCenter.JOB_SCHEDULER_MAP.computeIfAbsent(jobId, key -> new ConcurrentHashMap<>());
+        RuleAlteredJobScheduler scheduler = schedulerMap.get(shardingItem);
+        if (scheduler == null) {
```

Comment:
1, It's better to supply method in RuleAlteredJobSchedulerCenter to access needed RuleAlteredJobScheduler.

2, `scheduler == null` could be `null == scheduler`